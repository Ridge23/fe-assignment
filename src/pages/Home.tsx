import React from 'react';

import Layout from '../components/layout/Layout';
import Filter from '../components/filter/Filter';

import './sass/home.scss';
import ServiceListItem from '../components/services/ServiceListItem';

export default function Home(): JSX.Element {
    return (<Layout title="Services">
        <Filter />
        <div className="services-container">
            <ServiceListItem />
            <ServiceListItem />
            <ServiceListItem />
            <ServiceListItem />
        </div>
    </Layout>)
}