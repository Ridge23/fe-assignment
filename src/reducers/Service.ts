interface IService {
    id: number;
    title: string;
    description: string;
    promocode: string;
    isActive: boolean;
}
interface IServiceState {
    services: IService[];
    isFetching: boolean;
    errorCode?: number;
}

interface IServiceAction {
    type: string;
    services?: IService[];
    errorCode?: number;
}

const initState: IServiceState = {
    services: [],
    isFetching: false,
    errorCode: 0,
};

function Service(state = initState, action: IServiceAction): IServiceState {
    switch (action.type) {
        case 'REQUEST_SERVICES':
            return {
                ...state,
                isFetching: true,
            }
        case 'RECEIVE_SERVICES':
            return {
                ...state,
                isFetching: false,
                services: action.services,
            }
        case 'RECEIVE_SERVICES_FAILED':
            return {
                ...state,
                isFetching: false,
                errorCode: action.errorCode,
            }
        default:
            return state;
    }
}

export { Service as default, IServiceState, IService };