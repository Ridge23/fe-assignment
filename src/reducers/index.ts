import { combineReducers } from 'redux';

import Service from './Service';

const rootReducer = combineReducers({
    Service
})

export default rootReducer