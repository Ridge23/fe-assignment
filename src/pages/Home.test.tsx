import React from 'react';
import { shallow } from 'enzyme';

import Home from './Home';
import { IService } from 'reducers/Service';

const handleFetchServices = jest.fn();
const hadnleSetServiceActive = jest.fn();

describe('Home', () => {
    it('should render correctly with data', () => {
        const services: IService[] = [
            {
                id: 1,
                title: 'test_data',
                description: 'test_description',
                promocode: 'test_promocode',
                isActive: false
            },
            {
                id: 2,
                title: 'test_data_1',
                description: 'test_description_1',
                promocode: 'test_promocode_1',
                isActive: true
            },
        ]

        const component = shallow(<Home 
            services={services}
            getServices={handleFetchServices} 
            setServiceActive={hadnleSetServiceActive}
        />
        );

        expect(component.find('ServiceListItem').length).toBe(2);

        expect(component).toMatchSnapshot();
    });
});