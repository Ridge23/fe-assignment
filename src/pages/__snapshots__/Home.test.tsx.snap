// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly with data 1`] = `
<Layout
  title="Services"
>
  <Filter
    handleChange={[Function]}
  />
  <div
    className="services-container"
  >
    <ServiceListItem
      description="test_description"
      id={1}
      isActive={false}
      key="service-1"
      onClick={[Function]}
      promocode="test_promocode"
      title="test_data"
    />
    <ServiceListItem
      description="test_description_1"
      id={2}
      isActive={true}
      key="service-2"
      onClick={[Function]}
      promocode="test_promocode_1"
      title="test_data_1"
    />
  </div>
</Layout>
`;
